modules/class-simplechart-plugin-versions.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+/*
+ * This class manages when we show alerts to upgrade
+ */
+
+class Simplechart_Plugin_Versions {
+
+	private $_simplechart_releases_url = 'https://api.github.com/repos/alleyinteractive/wordpress-simplechart/releases';
+
+	private $_simplechart_url = 'https://github.com/alleyinteractive/wordpress-simplechart/';
+
+	private $_latest_plugin_version;
+	private $_latest_plugin_zip_url;
+	private $_latest_plugin_update_name;
+	private $_latest_plugin_update_description;
+	private $_latest_plugin_last_updated;
+
+	public function __construct() {
+		add_action( 'admin_init', array( $this, 'simplechart_check_for_new_version' ) );
+		add_filter( 'plugins_api_result', array( $this, 'simplechart_mock_plugins_api' ), 10, 3 );
+	}
+
+	/**
+	 * Compares remote version transient to current version checking for update
+	 */
+	public function simplechart_check_for_new_version() {
```

Comment:
Also, you don't need to namespace method names inside the class, i.e. this can be just `check_for_new_version()`